src/Microsoft.DotNet.Build.Tasks/Targets/encoding.targets

```diff
@@ -0,0 +1,46 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="GenerateEncodingTable" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll" />
+
+  <!--
+       Include the following in a project which needs to generate an encoding data-table, from the project directory:
+       <PropertyGroup>
+         <IANAMappingPath>$(MSBuildThisFileDirectory)\nameOfMappingFile</IANAMappingPath>
+         <PreferredIANANamesPath>$(MSBuildThisFileDirectory)\nameOfPreferredNameFile</PreferredIANANamesPath>
+       </PropertyGroup>
+       <Import Project="$(ToolsDir)encoding.targets" Condition="Exists('$(ToolsDir)encoding.targets')" />
+  -->
+
+  <PropertyGroup>
+    <IntermediateEncOutputFileFullPath>$(IntermediateOutputPath)EncodingData.Table.cs</IntermediateEncOutputFileFullPath>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(IANAMappingPath)'!='' And '$(PreferredIANANamesPath)'!=''">
```

Comment:
As I mentioned above lets not hook this into the default build, so I would remove this CompileDependsOn property update.